Fix aws oidc deployment metrics

bernardjkim opened the pull request:
This fixes Teleport updater metrics for AWS OIDC deployments.

The AWS OIDC deployments are not deployed with their own Teleport updater, instead updates are handled by the proxy instances. This means that those deployments do not have the `TELEPORT_EXT_UPGRADER` environment variables set even though they are being automatically updated.

This change adds the `TELEPORT_EXT_UPGRADER` to these deployments, which will ensure that these instances now report that they are enrolled in auto updates.

changelog: Fix Teleport updater metrics for AWS OIDC deployments

bernardjkim committed:
Add TELEPORT_EXT_UPGRADER env to AWS OIDC deployments

bernardjkim committed:
Add unit tests

bernardjkim committed:
Fix import

zmb3 commented on lib/integrations/awsoidc/deployservice_test.go:
```diff
@@ -209,3 +214,36 @@ func TestNormalizeECSResourceName(t *testing.T) {
+func TestUpsertTask(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("verify expected environment variables", func(t *testing.T) {
```
This is just a single test, so I don't see any value in making a subtest.

zmb3 commented on lib/integrations/awsoidc/deployservice_test.go:
```diff
@@ -19,14 +19,19 @@
 	"testing"
 
+	"github.com/aws/aws-sdk-go-v2/aws"
+	ecsTypes "github.com/aws/aws-sdk-go-v2/service/ecs/types"
```
```suggestion
	ecstypes "github.com/aws/aws-sdk-go-v2/service/ecs/types"
```
Package names (even aliased ones) should use all lower case.

bernardjkim committed:
Remove subtest

bernardjkim committed:
Remove constants pkg

bernardjkim committed:
Iterate over all container definitions

bernardjkim committed:
Merge branch 'master' into bernard/aws-oidc-metrics

bernardjkim committed:
Remove TELEPORT_EXT_UPGRADER from ensured env var

TELEPORT_EXT_UPGRADER=awsoidc_integration is not supported when
attempting to initialize a new exporter. Instead of including version
checks, it is safer to use the install method environment variable to
populate the upgrader value in the heartbeat.

bernardjkim committed:
Use constant

bernardjkim committed:
Remove comment
